test/decl/ext/generic.swift

```diff
@@ -132,7 +132,7 @@ func genericClassEquatable<T : Equatable>(_ gc: GenericClass<T>, x: T, y: T) {
 }
 
 func genericClassNotEquatable<T>(_ gc: GenericClass<T>, x: T, y: T) {
-  gc.foo(x, y: y) // expected-error{{type 'T' does not conform to protocol 'Equatable'}}
+  gc.foo(x, y: y) // expected-error{{argument type 'T' does not conform to expected type 'Equatable'}}
```

Comment:
Do we have two almost identical diagnostic strings? When is the former emitted?

Reply:
The former is from TypeChecker::diagnoseConformanceFailure() and it gets used for problems with derived conformances and actual you-declared-it-but-didn't-meet-the-requirements on declarations. It's also a sort of last resort error in a couple remaining spots in CSDiag.cpp, but I think the trend/intention is for those uses to eventually all go away.